### PR TITLE
Single Class Span Classification Support

### PR DESCRIPTION
**Pull request by Steven-Yiran**
## Summary

**This pull request aims to serve as the starting point to support span classification datasets proposed in #385. Currently, only single-class span classification is supported. One example of a single-class span classification task is [Extractive Question Answering](https://huggingface.co/learn/nlp-course/chapter7/7?fw=pt#question-answering).**

```python
# Example code snippet
import numpy as np
from cleanlab.experimental.span_classification import find_label_issues

labels = [[0, 0, 1, 1], [1, 1, 0]]
pred_probs = [
    np.array([0.9, 0.9, 0.9, 0.1]),
    np.array([0.1, 0.1, 0.9]),
]

issues = find_label_issues(labels, pred_probs)
```

## Example Usage

An example of using the `span_classification` module to find label issues in the extractive question-answering dataset [SQuAD](https://rajpurkar.github.io/SQuAD-explorer/) is provided in a separate repository [here](https://github.com/Steven-Yiran/example_spancat/tree/main).

### CODE REVIEW

File: cleanlab/experimental/span_classification.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    Tokens identified with issues will be ranked by their individual label quality score.
+
+    To rank the sentences based on their overall label quality, use :py:func:`experimental.span_classification.get_label_quality_scores <cleanlab.experimental.span_classification.get_label_quality_scores>`
+
+    Parameters
+    ----------
+    labels:
+        Nested list of given labels for all tokens, such that `labels[i]` is a list of labels, one for each token in the `i`-th sentence.
+
+        Currently, only a single span class is supported.
```
Comment (aditya1503) [inline cleanlab/experimental/span_classification.py:29]:
```suggestion
        Nested list of given labels for all tokens.
         Refer to documentation for this argument in :py:func:`token_classification.filter.find_label_issues <cleanlab.token_classification.filter.find_label_issues>` for further details.

        Currently, only a single span class is supported.
```

File: cleanlab/experimental/span_classification.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+    pred_probs:
+        List of np arrays, such that `pred_probs[i]` has shape ``(T, K)`` if the `i`-th sentence contains T tokens.
+
+        Each element of `pred_probs[i]` corresponds to a token `t` in the `i`-th sentence,
+        and contains model-predicted probabilities that `t` belongs to the span classes.
```
Comment (aditya1503) [inline cleanlab/experimental/span_classification.py:35]:
```suggestion
        An array of shape ``(T, K)`` of model-predicted class probabilities.
       Refer to documentation for this argument in :py:func:`token_classification.filter.find_label_issues <cleanlab.token_classification.filter.find_label_issues>` for further details.
```

**Comment by aditya1503** [comment]
Hi @Steven-Yiran,

Thank you for your valuable contributions. I noticed your approach involves obtaining predicted probabilities in a one-vs-rest format. Are you aligning with the strategy outlined in this GitHub thread (https://github.com/cleanlab/cleanlab/issues/385#issuecomment-1741676676)? Additionally, are you consolidating the results as suggested in the end?
Best
Aditya

File: tests/test_span_classification.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    find_label_issues,
+    display_issues,
+    get_label_quality_scores,
+)
+import numpy as np
+import pytest
+import warnings
+
+warnings.filterwarnings("ignore")
+words = words = [["I", "love", "Cleanlab", "Inc"], ["A", "new", "park"]]
```
Comment (aditya1503) [inline tests/test_span_classification.py:11]:
```suggestion
words = [["I", "love", "Cleanlab", "Inc"], ["A", "new", "park"]]
```

**Review by aditya1503 — APPROVED** [review]

**Review by aditya1503 — APPROVED** [review]

**Review by aditya1503 — APPROVED** [review]
Thanks a lot for your contributions @Steven-Yiran

**Comment by Steven-Yiran** [comment]
Thanks @aditya1503 @jwmueller for all the helpful guidance and modifications!

**Pull request merged**